frontend/ui/trapper.lua

```diff
@@ -674,7 +673,11 @@ function Trapper:dismissableRunInSubprocess(task, trap_widget_or_string, task_re
     end
     -- return what we got or not to our caller
     if ret_values then
-        return completed, unpack(ret_values)
+        if ret_n then
+            return completed, unpack(ret_values, 1, ret_n)
```

Comment:
unpack() has no issue with "key"=value (non idx) in the table ? and will just ignore your `ret_values.n` ? (just asking, no idea)

Reply:
`unpack` only operates on the array part of the table, so the n (or any other hash) are skipped. (This is why the actual return values are wrapped in an array).

Reply:
e.g., 

```
> t = { n = 2, [1] = 1, [2] = 2, foo = "bar" }
> print(unpack(t, 1, t.n))
1       2
> t = { foo = bar }
> print(unpack(t))

> 
```